### PR TITLE
Search enhance user preferences

### PR DESCRIPTION
**Pull request by dyampols**
- Search now takes into user preferences 
- If user is not logged in will search without preferences 
- If user does not have preferences it will search without preferences 
- all of these cases are tested 

TODO: 
- Add handling of no input in search

### CODE REVIEW

File: travvy/src/components/AttractionsList.vue [hunk 1]
```diff
@@ -42,7 +42,7 @@
       <div v-if="$store.state.recommendedAttractions.length > 0">
       <br>
       <h3>Description</h3>
-      <p>{{$store.state.city}} is the largest city in {{$store.state.recommendedAttractions[0].country}}, and is home to attractions such as the {{$store.state.recommendedAttractions[0].attraction_name}}, {{$store.state.recommendedAttractions[1].attraction_name}}, and {{$store.state.recommendedAttractions[2].attraction_name}}.</p>
+      <p>{{$store.state.city}} is the largest city in {{$store.state.recommendedAttractions[0].country}}, and is home to attractions such as the {{$store.state.recommendedAttractions[0].attraction_name}}.</p>
```
Comment (dyampols) [inline travvy/src/components/AttractionsList.vue:45]: This would not work if the return was of length less than one

File: test/unitTests.js [hunk 2]
```diff
@@ -78,17 +78,33 @@ describe('Correct email and correct password log in test', function() {
...
+              assert.equal(response.length,1);
+        })
+    });
+
+describe('Recommend Attractions User without Preferences Test', function() {
+        it('Recommend Attractions User without Preferences should return 6', async function() {
+              const response = await RecommendController.recommend(city2,groupSize,startDate,endDate,userEmailWithoutPreferences)
+              assert.equal(response.length,6);
+        })
+    });
```
Comment (emilynairnn) [inline test/unitTests.js:107]: Good idea to test the cases with preferences and without!

File: travvy/src/services/AttractionsService.js [hunk 3]
```diff
@@ -3,8 +3,8 @@ import axios from 'axios'
 
 export default {
   // It makes get request to recommend attractions based on city
-    recommend(city,groupSize,startDate,endDate) {
-        return axios.get('/recommend', {params:city,groupSize,startDate,endDate})  
+    recommend(city,groupSize,startDate,endDate,user) {
+        return axios.get('/recommend', {params:city,groupSize,startDate,endDate,user})
```
Comment (emilynairnn) [inline travvy/src/services/AttractionsService.js:7]: Interesting to see how the user field was added in!

**Review by emilynairnn — APPROVED** [review]
This looks great Daria! I think its an important feature in our application, and I am happy you were able to implement it this sprint.

File: test/unitTests.js [hunk 4]
```diff
@@ -78,17 +78,33 @@ describe('Correct email and correct password log in test', function() {
 // Unit Tests for Home Page
 const assert = require('assert');
 const city = "Toronto"
+const city2 = "Paris"
 const groupSize = 20
 const startDate = new Date("2020-11-05")
 const endDate = new Date("2021-01-05")
+const userEmailWithPreferences = 'dyampolsky@uwaterloo.ca'
+const userEmailWithoutPreferences = 'lucytest@test.ca'
```
Comment (lkdfount) [inline test/unitTests.js:86]: Happy to see we are using our test instances for what they were intended for!

File: travvy/src/components/AttractionsList.vue (hunk 1 above)
Comment (lkdfount) [reply travvy/src/components/AttractionsList.vue:45]: Good idea to change it. Hopefully in future sprints we can make this writing more dynamic.

**Review by lkdfount — APPROVED** [review]

File: travvy/src/components/AttractionsList.vue (hunk 1 above)
Comment (dyampols) [reply travvy/src/components/AttractionsList.vue:45]: Yes, I agree, I can add this as ticket in the backlog for the next sprint!

**Pull request merged**